apm-sniffer/apm-sdk-plugin/light4j-plugins/light4j-plugin/src/main/java/org/apache/skywalking/apm/plugin/light4j/HandleRequestInterceptor.java

```diff
@@ -0,0 +1,141 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *
+ */
+
+package org.apache.skywalking.apm.plugin.light4j;
+
+import com.networknt.exception.ExceptionHandler;
+import com.networknt.handler.MiddlewareHandler;
+import com.networknt.handler.OrchestrationHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.util.HeaderMap;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.context.CarrierItem;
+import org.apache.skywalking.apm.agent.core.context.ContextCarrier;
+import org.apache.skywalking.apm.agent.core.context.ContextManager;
+import org.apache.skywalking.apm.agent.core.context.ContextSnapshot;
+import org.apache.skywalking.apm.agent.core.context.tag.Tags;
+import org.apache.skywalking.apm.agent.core.context.trace.AbstractSpan;
+import org.apache.skywalking.apm.agent.core.context.trace.SpanLayer;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.EnhancedInstance;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.InstanceMethodsAroundInterceptor;
+import org.apache.skywalking.apm.agent.core.plugin.interceptor.enhance.MethodInterceptResult;
+import org.apache.skywalking.apm.network.trace.component.ComponentsDefine;
+
+import java.lang.reflect.Method;
+
+/**
+ * {@link HandleRequestInterceptor} creates an entry span before the execution of
+ * {@link com.networknt.exception.ExceptionHandler#handleRequest(HttpServerExchange)} in the I/O thread.
+ *
+ * If the {@link Config.Plugin.Light4J#TRACE_HANDLER_CHAIN} flag is set, additionally a local span is produced for each
+ * {@link com.networknt.handler.MiddlewareHandler} and business handler before their respective
+ * {@link com.networknt.handler.LightHttpHandler#handleRequest(HttpServerExchange)} method executes.
+ * Since {@link com.networknt.handler.LightHttpHandler} is implemented by various middleware and business handlers and
+ * the Light4J framework delegates to these in succession, a chain of
+ * {@link org.apache.skywalking.apm.agent.core.context.trace.LocalSpan}s will be produced.
+ *
+ * @author tsuilouis
+ */
+public class HandleRequestInterceptor implements InstanceMethodsAroundInterceptor {
+
+    @Override
+    public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
+                             MethodInterceptResult result) {
+        if (isExceptionHandler(objInst)) {
+            HttpServerExchange exchange = (HttpServerExchange) allArguments[0];
+
+            if (exchange.isInIoThread()) {
+
+                String operationName = exchange.getRequestPath() + "@" + exchange.getRequestMethod();
```

Comment:
Could you give me an example about this? How does the op name look like?

Reply:
The op name looks like "/v1/pets@GET". This is the format that light-4j uses with Jaeger implementation of OpenTracing: https://github.com/networknt/light-4j/blob/master/jaeger-tracing/src/main/java/com/networknt/jaeger/tracing/JaegerHandler.java#L79.

I would like to keep the HTTP verb in the op name, but I am open to changing to the rest of the format.

Reply:
In existing SpringMVC, the format looks like `{PUT, GET} /a/b/c`, based on `RequestMapping` settings. I would recommend we use the same format in light4j

https://github.com/apache/skywalking/blob/c694a30c5fc66dd576c331738ed3bfde760babfe/apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/RequestMappingMethodInterceptor.java#L48-L57

Reply:
Just to clarify using your example, if the request sent is `GET /a/b/c` you would like to see  the op name as just `{GET} /a/b/c` and not `{PUT, GET} /a/b/c`, correct?

Reply:
In SpringMVC, it is config based, so, if one method could support PUT and GET, then it has {PUT, GET} in OP name. What is the case in Light4j?

Reply:
In spring, one controller can handle multiple requests with different HTTP verbs and I think it makes sense to have the format that supports multiple verbs. Light-4j is targeted to microservices and each endpoint will have only one verb and path combined and list of middleware handlers and business handlers. As we are supporting other tracing product like Jaeger and planning to support others. It makes sense to use the Light-4j endpoint for us as our users are familiar with the path@verb formation. What do you think?

Reply:
In today's analysis, we don't care about the format much. I am just thinking to make name format consistently. That is the only reason I asked for `{GET} /a/b/c`.

Reply:
I assume that the op name is part of the span that will be displayed on the UI to identify which endpoint the request is coming from. If this assumption is right, then I think the familiarity of the format from framework users is more important than the format itself.

Reply:
I think both points are valid to the question of "What span op name format does the user of Skywalking want to see?". If the user prefers the Skywalking platform's format, then they would expect `{GET} /a/b/c`. Other users may prefer the format to be consistent with respect to the framework they use. Perhaps we should have another config option to let the user decide which format they would like to see? I believe this has been done before with one of the Spring plugins having a flag to pick the format for the operation name. The default can follow the Skywalking format.

Reply:
If you are sure your(light4j) users are familiar with the current format(with `@`), I am OK to don't change it to SpringMVC's.